test/test_reductions.py

```diff
@@ -293,6 +294,96 @@ def test_empty_tensor_nonempty_slice(self, device, op: ReductionOpInfo):
             result = op(t, *args, dim=dim, **kwargs)
             self.assertEqual(result.shape, _reduced_shape(t.shape, dim))
 
+    def _test_noncontiguous(self, op: ReductionOpInfo, t: torch.Tensor, **reduction_kwargs):
+        """Helper method to test noncontiguous input tensors."""
+        assert not t.is_contiguous()
+
+        t_contig = t.contiguous()
+        for args, kwargs in op.generate_args_kwargs(t_contig, **reduction_kwargs):
+            kwargs.update(reduction_kwargs)
+            result = op(t, *args, **kwargs)
+            expected = op(t_contig, *args, **kwargs)
+            self.assertEqual(result, expected)
+
+    @ops(reduction_ops)
+    def test_noncontiguous_innermost(self, device, dtype, op: ReductionOpInfo):
+        """Tests reducing along noncontiguous innermost dimension."""
+        t = make_tensor((10, 10), device, dtype)
```

Comment:
make_tensor has a noncontiguous kwarg

https://github.com/pytorch/pytorch/blob/fcc1f87b6aacdfba9193691568bc405be42ac77e/torch/testing/_internal/common_utils.py#L1942

Reply:
I guess the issue with using that is you want to control the noncontiguity precisely here

Reply:
Yeah, in this case I'd like to test for different types of non-contiguity